### ISSUE TITLE
Allow site-wide credentials

### ISSUE THREAD

**Issue by strugee** [cantino/huginn#1592]
Use-case: I've set up a Jabber account, `huginn@strugee.net`, that I use to get realtime alerts for stuff. I know the password, because I have root on the server, but no one else can use the account without me handing out the password (unideal, obviously). I'd like to set up some sort of site-wide credential with the password, so that people can use the password with the Jabber agent, but I'm not just _giving_ them access to the entire Jabber account.

Note that for this to be useful, Huginn would have to make sure the user never sees the credential. That means making sure it doesn't show up in e.g. event bodies, and probably no JS agent access. Obviously those requirements make this more complicated.

Thoughts?

**Comment by cantino** [comment]
That seems reasonable.  Maybe Credentials could have a `shared` checkbox that any `admin = true` user can check?  When checked, all users see the credential (but cannot view / edit the value).

**Comment by dsander** [comment]
I agree that sounds really tricky to hide everywhere. Expanding the services could work well, Agents would "register" themselves with the needed authentication options (password, username/password, etc), and the Services page would then show the required fields much like it is done for `FormConfigurable`.

**Comment by strugee** [comment]
@dsander so to clarify, you're suggesting that e.g. the Jabber agent would register "I need a username and password for a Jabber account," and then I could go into the services page and fill those values?

**Comment by dsander** [comment]
@strugee Right, we probably would need to extend the Service model, but that way the credentials should be save as only the Agent using the Service would have access to them.

**Comment by albertsun** [comment]
Maybe this is a case where the agent should be shared instead of the credential or service?

Then you create one jabber agent in your account that has access to the credentials, mark it as shared, and it can show up as a source for everyone else's agents so they can receive events without them being able to edit the agent.

**Comment by cantino** [comment]
That's interesting @albertsun.
